db.js

```diff
@@ -539,10 +576,9 @@ exports.init = function (sbot, config) {
       indexName = 'base'
     }
 
-    // setTimeout to make sure extra indexes from secret-stack are also included
-    setTimeout(() => {
+    secretStackLoaded.onReady(() => {
```

Comment:
Notice that the `setTimeout` was causing a forced async wait for **every** onDrain call we did, even if the setTimeout was useful only during init (to wait for secret-stack plugins to register indexes). This new method (using `ReadyGate`) means that if it's `ready`, it'll be called immediately/synchronously. I expect this will improve overall all ssb-db2 queries a tiny bit.

Reply:
Yes, great find